src/loctool/PendoXliffFileType.ts

```diff
@@ -0,0 +1,347 @@
+        this.loctoolAPI = loctoolAPI;
+    }
+
+    /**
+     * Reference to the loctool {@link Project} instance which uses this file type.
+     */
+    private readonly project: Project;
+
+    /**
+     * Additional functionality provided by loctool to the plugin.
+     */
+    private readonly loctoolAPI: API;
+
+    private static readonly extensions = [".xliff", ".xlf"];
+    getExtensions(): string[] {
+        return PendoXliffFileType.extensions;
+    }
+
+    /** human-readable file type name */
+    private static readonly name = "Pendo XLIFF";
+    name(): string {
+        return PendoXliffFileType.name;
+    }
+
+    /**
+     * [XLIFF datatype](https://docs.oasis-open.org/xliff/v1.2/os/xliff-core.html#datatype)
+     * identifier for Pendo markdown strings.
+     *
+     * XLIFFs exported from pendo use <file datatype="pendoguide">,
+     * so resources produced by this plugin are custom escaped versions of that datatype.
+     */
+    public static readonly datatype = "x-pendoguide-escaped";
+    getDataType(): string {
+        // strings in XLIFFs exported from Pendo are markdown with custom extensions
+        return PendoXliffFileType.datatype;
+    }
+
+    getDataTypes(): Record<string, string> | undefined {
+        // use defaults
+        return undefined;
+    }
+
+    get sourceLocale() {
+        // Per convention (e.g. https://github.com/iLib-js/loctool/blob/285401359f923c1be11e7329b549ed11b4099637/lib/MarkdownFileType.js#L70)
+        // it seems that source locale should always come from the project
+        return this.project.getSourceLocale();
+    }
+
+    handles(pathInProject: string): boolean {
+        // only pick files which match some path mapping
+        const pathMapping = this.getMappingForPath(pathInProject);
+        if (!pathMapping) {
+            return false;
+        }
+
+        // per convention, only pick files which match project's source locale
+        // (this is done to avoid processing files which are already localized)
+        /* eslint-disable-next-line
+         @typescript-eslint/no-unsafe-assignment,
+         @typescript-eslint/no-unsafe-call,
+         @typescript-eslint/no-explicit-any,
+         @typescript-eslint/no-unsafe-member-access -- yet undocumented utility method,
+         signature based on https://github.com/iLib-js/ilib-loctool-tap-i18n/blob/7585e97497e16475bfce1fc034caf0c7716229e1/YamlFileType.js#L122 */
+        const fileLocale: string = (this.loctoolAPI.utils as any).getLocaleFromPath(
+            pathMapping.template,
+            pathInProject,
+        );
+        if (fileLocale !== this.sourceLocale) {
+            return false;
+        }
+
+        return true;
+    }
+
+    write(): void {
+        // no-op
+        // per loctool convention, when localized files are written individually
+        // there is no need to implement this method
+        // (which is meant to write out aggregated resources - whatever that means);
```

Comment:
There are 2 general modes of operation for the loctool:

1. Take a source file, extract the strings from it, localized them, then write out a translated copy of that file. For this mode, the output file type is the same as the source file.
2. Extract strings from source files, gather them together into a resource file, localize the strings in that file, and write out the translated resource files. For this mode, the output file type is different than the source file.

"Aggregated resources" means the 2nd mode. Example: swift files have the strings in them wrapped with NSTranslatedString(). These are extracted from the .swift files, aggregated together with strings from all the other swift files, localized, then written out to a .strings resource file.

Reply:
Ok so this is not needed for this plugin correct?

Frankly I don't understand why this is a separate plugin interface method, since from what you're saying this could just be
```js
localize(translations, locales) {
  fs.writeFileSync(
    'path/to/accumulated/strings/file',
    makeStringsFile(
      translations.getAll().filter(r => locales.includes(r.getTargetLocale()))
    )
  );
}